vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts

```diff
@@ -26,16 +33,101 @@ import
 } from './DotnetInstall';
 import { IAcquisitionWorkerContext } from './IAcquisitionWorkerContext';
 import { IInstallationDirectoryProvider } from './IInstallationDirectoryProvider';
-import { InstallRecord, InstallRecordOrStr } from './InstallRecord';
+import { InstallOwner, InstallRecord, InstallRecordOrStr } from './InstallRecord';
 
 export type InstallState = 'installing' | 'installed';
 
+export type SessionsWithInUseExecutables = Map<string, Set<string>>;
+
 export class InstallTrackerSingleton
 {
     protected static instance: InstallTrackerSingleton;
+    protected static sessionId: string;
```

Comment:
It seems like we would want to not have this be static, since this bypasses the constructor which instantiates the id.